pkg/eigenState/operatorDirectedRewardSubmissions/operatorDirectedRewardSubmissions.go

```diff
@@ -0,0 +1,334 @@
+}
+
+func parseRewardSubmissionOutputData(outputDataStr string) (*operatorDirectedRewardSubmissionOutputData, error) {
+	outputData := &operatorDirectedRewardSubmissionOutputData{}
+	decoder := json.NewDecoder(strings.NewReader(outputDataStr))
+	decoder.UseNumber()
+
+	err := decoder.Decode(&outputData)
+	if err != nil {
+		return nil, err
+	}
+
+	return outputData, err
+}
+
+func (odrs *OperatorDirectedRewardSubmissionsModel) handleOperatorDirectedRewardSubmissionCreatedEvent(log *storage.TransactionLog) ([]*OperatorDirectedRewardSubmission, error) {
+	arguments, err := odrs.ParseLogArguments(log)
+	if err != nil {
+		return nil, err
+	}
+
+	outputData, err := parseRewardSubmissionOutputData(log.OutputData)
+	if err != nil {
+		return nil, err
+	}
+	outputRewardData := outputData.OperatorDirectedRewardsSubmission
+
+	rewardSubmissions := make([]*OperatorDirectedRewardSubmission, 0)
+
+	for i, strategyAndMultiplier := range outputRewardData.StrategiesAndMultipliers {
+		startTimestamp := time.Unix(int64(outputRewardData.StartTimestamp), 0)
+		endTimestamp := startTimestamp.Add(time.Duration(outputRewardData.Duration) * time.Second)
+
+		multiplierBig, success := numbers.NewBig257().SetString(strategyAndMultiplier.Multiplier.String(), 10)
+		if !success {
+			return nil, xerrors.Errorf("Failed to parse multiplier to Big257: %s", strategyAndMultiplier.Multiplier.String())
+		}
+
+		for j, operatorReward := range outputRewardData.OperatorRewards {
+			amountBig, success := numbers.NewBig257().SetString(operatorReward.Amount.String(), 10)
+			if !success {
+				return nil, xerrors.Errorf("Failed to parse amount to Big257: %s", operatorReward.Amount.String())
+			}
+
+			rewardSubmission := &OperatorDirectedRewardSubmission{
+				Avs:             strings.ToLower(arguments[1].Value.(string)),
+				RewardHash:      strings.ToLower(arguments[2].Value.(string)),
+				Token:           strings.ToLower(outputRewardData.Token),
+				Operator:        strings.ToLower(operatorReward.Operator),
+				OperatorIndex:   uint64(j),
+				Amount:          amountBig.String(),
+				Strategy:        strings.ToLower(strategyAndMultiplier.Strategy),
+				StrategyIndex:   uint64(i),
+				Multiplier:      multiplierBig.String(),
+				StartTimestamp:  &startTimestamp,
+				EndTimestamp:    &endTimestamp,
+				Duration:        outputRewardData.Duration,
+				BlockNumber:     log.BlockNumber,
+				TransactionHash: log.TransactionHash,
+				LogIndex:        log.LogIndex,
+			}
+
+			rewardSubmissions = append(rewardSubmissions, rewardSubmission)
+		}
+	}
+
+	return rewardSubmissions, nil
+}
+
+func (odrs *OperatorDirectedRewardSubmissionsModel) GetStateTransitions() (types.StateTransitions[[]*OperatorDirectedRewardSubmission], []uint64) {
+	stateChanges := make(types.StateTransitions[[]*OperatorDirectedRewardSubmission])
+
+	stateChanges[0] = func(log *storage.TransactionLog) ([]*OperatorDirectedRewardSubmission, error) {
```

Comment:
Out of curiousity. Why are we doing this for block `0`? Still haven't been to understand this bit.

Reply:
The idea here is that you could potentially have different versions of the state transition function over time as a state model evolves. When picking a version to use, we iterate through a reverse sorted list of the block numbers (keys) in `stateChanges` and pick the one where `blockNumberKey >= currentBlockNumber`.

This way you can always sync from scratch since the handlers are versioned by block height

Reply:
So I'm assuming the `0` is just hardcoded for now to pick the first version?

Reply:
Correct